feat: add ExtraMojo

sstadick opened the pull request:
Checklist
- [X] License file is packaged (see [here](https://github.com/modular/modular-community/blob/dbe0200598733fea411ee2246507705e8ea07a32/recipes/hue/recipe.yaml#L33-L40) for an example).
- [X] Set the build number to `0` (for new packages, or if the version changed)
- [ ] Bumped the build number (if the version is unchanged)


This adds the ExtraMojo package, which has functionality for working with files and strings that isn't in the standard library yet.

`rattler-build build -c https://conda.modular.com/max -c conda-forge --skip-existing=all -r ./recipe.yaml` worked locally.

carolinefrasca commented:
@sstadick Thank you for submitting this! Can you please update the `package/name` field in the [`recipe.yaml` file](https://github.com/modular/modular-community/pull/40/files) to your package name?

sstadick commented:
@carolinefrasca updated!

sstadick commented:
@carolinefrasca - I'm not sure what is going on in CI. It looks like it's trying to build mojmelo and failing, and also failing to build my package. But weirder, it's trying to build both of those on 24.5, when they both specify 24.6. I can't get that to reproduce locally.

carolinefrasca commented:
Hi @sstadick, sorry for the confusion. We resolved the issue with mojmelo by specifying the MAX version. To prevent the CI from building your package for 24.5 (I'm assuming you only want it to build for 24.6), can you change line 18 in your `recipe.yaml` file to be ` - max ==24.6`?

sstadick commented:
I've updated it to 24.6 👍

carolinefrasca commented:
Thank you so much @sstadick! We're still seeing this error:
```
Error:   × failed to setup test environment: Cannot solve the request because of: No
  │ candidates were found for extramojo ==0.6.0 he2932c6_0.
  │ 
  ╰─▶ Cannot solve the request because of: No candidates were found for
      extramojo ==0.6.0 he2932c6_0.
```
I suspect we can resolve this by adding this to the line 29 of your `recipe.yaml`:
```
requirements:
      run:
        - max ==24.6
```

Sorry for the additional ask!

sstadick commented:
@carolinefrasca - updated! Also updated ExtraMojos project file to pin the max version there, and updated the recipe to use the new version (0.7.0) that is pinned.

sstadick commented on recipes/ExtraMojo/recipe.yaml:
```diff
@@ -0,0 +1,54 @@
+    - mojo package ExtraMojo -o ${{ PREFIX }}/lib/mojo/ExtraMojo.mojopkg
+requirements:
+  host:
+    - max ==24.6
```
```suggestion
    - max =24.6
```

sstadick commented on recipes/ExtraMojo/recipe.yaml:
```diff
@@ -0,0 +1,54 @@
+  host:
+    - max ==24.6
+  run:
+    - max ==24.6
```
```suggestion
    - max =24.6
```

sstadick commented on recipes/ExtraMojo/recipe.yaml:
```diff
@@ -0,0 +1,54 @@
+        then:
+          - mojo run tests/test_file.mojo
+          - mojo test tests/test_regex.mojo
+          - mojo test tests/test_bstr.mojo
+    files:
```
```suggestion
          - mojo test tests/test_bstr.mojo
    requirements:
      run:
        - max =24.6
    files:
```

sstadick commented:
@carolinefrasca - I tried running the build locally without the `tests` copied in using the same test commands I use in the project but couldn't get it to work :( So I left the tests for now, because I did solve the other issue.

There seems to be something going on with the project names. I had it as `ExtraMojo`, but the build would fail when looking for an `extramojo` project. So I lowercased the project name and now it works.

carolinefrasca commented:
Thank you @sstadick! It looks like the only thing left is trailing whitespace in `recipe.yaml` – do you mind removing that?
